### PR TITLE
fix: increase timeout when writing screenshots

### PR DESCRIPTION
**Pull request by iProdigy**
Default write timeout is 10 seconds. This timeout can be breached for users with slow upload speeds trying to write screenshots of high resolution or detailed (e.g., with HD plugin) environments. This PR raises the write timeout for such calls to 30 seconds.

### CODE REVIEW

File: src/main/java/dinkplugin/message/DiscordMessageHandler.java [hunk 1]
```diff
@@ -33,14 +34,34 @@
...
+        this.executor = executor;
+        this.httpClient = httpClient.newBuilder()
+            .addInterceptor(chain -> {
+                Request request = chain.request().newBuilder()
+                    .header("User-Agent", DinkPlugin.USER_AGENT)
+                    .build();
+                Interceptor.Chain updatedChain = chain;
+                // Allow longer timeout when writing a screenshot file to overcome slow internet speeds
+                if (request.body() instanceof MultipartBody && Utils.hasImage((MultipartBody) request.body())) {
+                    updatedChain = chain.withWriteTimeout(30, TimeUnit.SECONDS); // default is 10 seconds
```
Comment (iProdigy) [inline src/main/java/dinkplugin/message/DiscordMessageHandler.java:58]: Should we get the timeout seconds from a hidden config setting instead?

Comment (pajlada) [reply src/main/java/dinkplugin/message/DiscordMessageHandler.java:58]: Yeah I think that would good, it adds some automatic documentation for the values floating around

**Review by pajlada — APPROVED** [review]
Looks good to me, yeah I think using a hidden config for this would be smart.

I imagine potential pluginhub blockers for this would be
a) Modifying the User Agent
b) Not using the standard HTTP client - I know creating your own HTTP client is frowned upon but this just modifies the request in flight a bit so I imagine it should be ok

**Comment by Mm2PL** [comment]
What if it was a normal config, visible in the ui?

19 gru 2022 08:14:39 iProdigy ***@***.***>:

**Comment by pajlada** [comment]
If we decide to expose this as a user-facing setting, I think it should be under a collapsed-by-default category. I'm not against an "Advanced" category with various useful-for-very-few-people settings

**Review by pajlada — APPROVED** [review]

**Comment by pajlada** [comment]
The latest commit adds an "Advanced" category at the bottom, which is collapsed by default.
The category itself has a tooltip with a warning for users
advanced-warning

It also exposes the retry options from #94 
advanced

The tooltip for the Image Upload Ti
advanced-image-upload-timeout_000
meout explains exactly what it does

**Pull request merged**